app/controllers/admin/court_application_hearing_controller.rb

```diff
@@ -0,0 +1,30 @@
+module Admin
+  class CourtApplicationHearingController < Admin::ApplicationController
+    def create
+      hearing = Hearing.find(court_application_params[:hearing_id])
+      court_application = CourtApplication.find(court_application_params[:court_application_id])
+
+      CourtApplicationHearing.find_or_create_by!(
+        court_application:,
+        hearing:,
+      )
+
+      redirect_to admin_court_application_path(court_application), notice: "Prosecution hearing was successfully add to Court application."
+    end
+
+    def destroy
+      court_application = CourtApplication.find(court_application_params[:court_application_id])
+      court_application_hearing = CourtApplicationHearing.find_by(hearing_id: court_application_params[:hearing_id], court_application_id: court_application_params[:court_application_id])
+
+      court_application_hearing.destroy!
+
+      redirect_to admin_court_application_path(court_application), notice: "Court application hearing was successfully deleted."
```

Comment:
If you replace `court_application` with `court_application_params[:court_application_id]` here, it saves you a DB query as you can delete line 16 entirely.